### PR TITLE
docs(geo): updates from DA feedback

### PR DESCRIPTION
**Pull request by TreTuna**
_Description of changes:_

- docs(geo): add info about map.remove
- docs(geo): add import in snippets
- docs(geo): all long, lat comments to reinforce order
- docs(geo): add geo to js configuration page
- docs(geo): add link to CLI search index provisioning

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by thaddmt — APPROVED** [review]

File: src/fragments/lib/geo/js/maps.mdx [hunk 1]
```diff
@@ -160,6 +162,15 @@ map.setStyle("myAmplifyGeoEsriTopographicMap"); // map name received from getAva
 map.resize(); // forces the map to re-render
 ```
 
+## Removing a map from the DOM
+
+When it's time to remove the map from the DOM, you can use the `.remove` method of the generated map. This will clean up and release all resources associated with the map (DOM elements, event bindings, web workers, and WebGL resources).
+Note: Afterwards, you must not call any other methods on the map.
```
Comment (siegerts) [inline src/fragments/lib/geo/js/maps.mdx:168]: Will something destructive happen if a method is called? Or, is it just that the method will not work? Just trying to gauge if this should be bolded or called out more as a `warning`.

Comment (aspittel) [reply src/fragments/lib/geo/js/maps.mdx:168]: Yeah, I think it's important to call out that if you don't do this in a framework like React you'll have a memory leak

**Comment by aspittel** [comment]
Hey, one piece of feedback throughout is to add more explanations before code snippets rather than just adding comments to the code, often they scroll off the page. I think this is a good first step though.

**Comment by sammartinez** [comment]
> Hey, one piece of feedback throughout is to add more explanations before code snippets rather than just adding comments to the code, often they scroll off the page. I think this is a good first step though.

Im in agreement with this with @aspittel. We should be give as much data as we can for our customers on this.

File: src/fragments/lib/geo/js/maps.mdx (hunk 1 above)
Comment (TreTuna) [reply src/fragments/lib/geo/js/maps.mdx:168]: > Will something destructive happen if a method is called? Or, is it just that the method will not work? Just trying to gauge if this should be bolded or called out more as a `warning`.

I'm actually not positive _what_ happens here, that's all the information MapLibre gives in their documentation about `remove()`, would need to do some manual messing around to know more atm.

Happy to update this more as a warning and add something specific calling out React for memory leaks.

https://maplibre.org/maplibre-gl-js-docs/api/map/#map#remove

image

**Comment by TreTuna** [comment]
> Hey, one piece of feedback throughout is to add more explanations before code snippets rather than just adding comments to the code, often they scroll off the page. I think this is a good first step though.

I understand this, but I believe it's also a balance upon doing this and also staying consistent with all the other categories in the documentation. 

In this case, I'm guessing the comment is mainly in reference with the "Top level configuration" portion, which is staying consistent by using comments as all of the other configuration objects on that page.

**Comment by TreTuna** [comment]
Made several updates around `.remove` and for React users:

The initial snippet to create a map now has a React section:
image
image

And the section on `.remove` now uses `<Callout warning>` for both:
- The problem of calling methods after removing the map
- Warning React users about memory leaks if they do not remove the map

File: src/fragments/lib/geo/js/maps.mdx (hunk 1 above)
Comment (TreTuna) [reply src/fragments/lib/geo/js/maps.mdx]: Added updates you can see below around this.

File: src/fragments/lib/geo/js/maps.mdx [hunk 2]
```diff
@@ -160,6 +214,26 @@ map.setStyle("myAmplifyGeoEsriTopographicMap"); // map name received from getAva
...
+## Removing a map from the DOM
+
+When it's time to remove the map from the DOM, you can use the `.remove` method of the generated map. This will clean up and release all resources associated with the map (DOM elements, event bindings, web workers, and WebGL resources).
+
+```javascript
+map.remove();
+```
+
+<Callout warning>
+After calling `.remove()`, you must not call any other methods on the map.
```
Comment (siegerts) [inline src/fragments/lib/geo/js/maps.mdx:226]:
```suggestion

After calling `.remove()`, you must not call any other methods on the map.

```

Comment (siegerts) [reply src/fragments/lib/geo/js/maps.mdx:226]: Just so that md renders as code

**Review by siegerts — APPROVED** [review]
small format on the callout for mdx to recognize the code hightlight. LGTM 🗺️

**Pull request merged**